### PR TITLE
`forum-id`: add automatically adding link, add dynamic enable/disable

### LINKED ISSUE #2468: Link to quoted post
**Is your feature request related to a problem? Please describe.**

Scratch doesn't provide an easy way to jump to the quoted post from a quote.

**Describe the solution you'd like**

An option in the Better quoter addon to automatically add a link when quoting a post.

**Describe alternatives you've considered**

It could also be its own addon.

### PR DESCRIPTION
**Pull request by CST1229**
Resolves #2468

### Changes

Adds an option in `forum-id` to automatically add a link to the post when it is quoted.
Also adds dynamic enable and disable to it.
(Maybe needs discussion: should we use post number or some other text?)

### Reason for changes

More convienient than having to click another link every time.

### Tests

Tested on Firefox.

(Actually real PR this time)

### CODE REVIEW

**Comment by WorldLanguages** [comment]
> (Maybe needs discussion: should we use post number or some other text?)

What are the alternatives?

**Comment by CST1229** [comment]
> 
> 
> > (Maybe needs discussion: should we use post number or some other text?)
> 
> What are the alternatives?

Something like "Original post" or "view post".

**Comment by RedGuy12** [comment]
Or make it a setting


________________________________

**Comment by CST1229** [comment]
> Or make it a setting

Done

File: addons/forum-id/userscript.js [hunk 1]
```diff
@@ -8,10 +8,41 @@ export default async function ({ addon, console, msg }) {
...
+    $.ajax("/discuss/post/" + id.substr(1) + "/source/").done(function (data) {
+      paste(
+        "[quote=" +
+          username +
+          "][small](" +
+          getIDLink(
+            id.substring(1),
+            addon.settings.get("auto_add_use_number")
+              ? post["0"].querySelector(".box-head > .conr").textContent
+              : "View post",
```
Comment (WorldLanguages) [inline addons/forum-id/userscript.js:39]: "View post" would need to be localized here (kind of - the forums themselves aren't localized completely).

**Comment by WorldLanguages** [comment]
@CST1229 To avoid having to localize "view post", could we just use the post number as originally intended, instead of making it togglable?

**Comment by CST1229** [comment]
> @CST1229 To avoid having to localize "view post", could we just use the post number as originally intended, instead of making it togglable?

Oh yeah I completely forgot about l10n.

**Comment by WorldLanguages** [comment]
I think it's better to make it always use the post number.

**Comment by CST1229** [comment]
> I think it's better to make it always use the post number.

Yes I thought that too.

**Comment by CST1229** [comment]
> I think it's better to make it always use the post number.

Done

**Pull request merged**